File: Golang/Ugly Number.go
//1
func isUgly(n int) bool {
    for n >= 1 {
        if n % 2 == 0 {
            n = n / 2
        } else if n % 3 == 0 {
            n = n / 3
        } else if n % 5 == 0 {
            n = n / 5
        } else if n == 1 {
            return true
        } else {
            return false
        }
    }

    return false
}

//2
func isUgly(num int) bool {
    if num <=0{
        return false
    }
    
    for num!=1{
        if num%2==0{
            num/=2
        }else if num%3==0{
            num/=3
        }else if num%5==0{
            num/=5
        }else{
            return false
        }
    }
    
    return true
}

//3
func isUgly(n int) bool {
    if n <= 0 {
        return false
    }

    for n % 5 == 0 {
        n /= 5
    }
    for n % 3 == 0 {
        n /= 3
    }
    for n % 2 == 0 {
        n /= 2
    }

    return n == 1
}

File: Python3/Ugly Number.py
#1
class Solution:
    def isUgly(self, n: int) -> bool:
        for p in 2, 3, 5:
            while n % p == 0 < n:
                n /= p
        return n == 1

#2
class Solution:
    def isUgly(self, n: int) -> bool:
        while n >= 1:
            if n%2 == 0:
                n=n//2
            elif n%3 == 0:
                n=n//3
            elif n%5 == 0:
                n=n//5
            elif n == 1:
                return True
            else:
                return False